lib/beaker/host/unix/file.rb

```diff
@@ -29,4 +29,201 @@ def file_exist?(path)
+  # @param [String] repo_name Name of the repo to check for the package in
+  #
+  # @return [String, nil] Path to the package repo if it exists, nil otherwise
+  def repo_path_exists(buildserver_url, package_name, build_version, repo_name)
+    variant, version, arch, codename = self['platform'].to_array
+    variant = 'el' if variant == 'centos'
+
+    repo_path = nil
+    case variant
+    when /fedora|el|centos/
+      fedora_prefix = ((variant == 'fedora') ? 'f' : '')
+
+      link =  "%s/%s/%s/repos/%s/%s%s/%s/%s/" %
+        [ buildserver_url, package_name, build_version, variant,
+          fedora_prefix, version, repo_name, arch ]
+      repo_path = link if link_exists?( link )
+    when /debian|ubuntu|cumulus/
+      path_candidate = "/root/#{package_name}/#{codename}/#{repo_name}"
+      repo_check = exec(
+        Beaker::Command.new("[[ -d #{path_candidate} ]]"),
+        :acceptable_exit_codes => [0,1] )
+      repo_path = path_candidate if repo_check.exit_code == 0
+    end
+    return repo_path
+  end
+
+  # Gets the development build repos list
+  #
+  # @param [Array<String>] build_repos custom repos to use before our defaults
+  #
+  # @return [Array<String>] Build repo array, fully flushed out
+  def dev_build_repos(build_repos = [])
+    package_repos = build_repos.nil? ? [] : [build_repos]
+    package_repos.push(['products', 'devel']) if self['platform'] =~ /fedora|el-|centos/
+    package_repos.flatten
+  end
+
+  # Gets the path to the repo for the given package
+  #
+  # @param [Array<String>] build_repos names of the repos to check for the package
+  # @param [String] buildserver_url URL to the buildserver
+  # @param [String] package_name Name of the package
+  # @param [String] build_version Version of the package
+  #
+  # @raise [RuntimeError] if not able to find a package repo on EL platforms
+  #
+  # @note On Debian-based systems, will default to the 'main' repo, unlike
+  #   EL-based ones, which will raise an error
+  #
+  # @return [String] Path to the repo for the given package
+  def repo_path(build_repos, buildserver_url, package_name, build_version)
+    repo_path = nil
+    package_repos = self.dev_build_repos( build_repos )
+    logger.trace("FOSSUtils#install_repo package repos: '#{package_repos}'")
+    package_repos.each do |repo|
+      repo_path = self.repo_path_exists(
+        buildserver_url, package_name, build_version, repo )
+      if repo_path
+        logger.debug("found repo at #{repo}:#{repo_path}")
+        break
+      else
+        logger.debug("couldn't find link at #{repo}, falling back to next option...")
+      end
+    end
+
+    case self['platform']
+    when /fedora|el-|centos/
+      raise "Unable to reach a repo directory at #{repo_path}" unless link_exists?( repo_path )
+    when /debian|ubuntu|cumulus/
+      if repo_path.nil?
+        repo_path = 'main'
+        logger.debug("using default repo '#{repo_path}'")
+      end
+    end
+
+    repo_path
+  end
```

Comment:
It looks like `repo_path_exists`, `dev_build_repos` and `repo_path` are dead code since we're no longer copying the puppet-agent repo to the local SUT and rewriting the repo_config to point to the local filesystem.